server/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesFetcher.java

```diff
@@ -117,6 +117,13 @@ static Map<String, IndexFieldCapabilities> retrieveFieldCaps(
         for (String field : fieldNames) {
             MappedFieldType ft = context.getFieldType(field);
             if (filter.test(ft)) {
+                Set<String> supportedAggregations;
+                if (ft.isAggregatable()) {
+                    var valuesSourceType = context.getForField(ft, MappedFieldType.FielddataOperation.SEARCH).getValuesSourceType();
```

Comment:
Maybe not load index field data? The only reason that index field data is loaded, is to figure out the value source type. Maybe `getValueSourceType()` method can be added to `MappedFieldType`? This avoids having to load field data. For some fields like _id this causes warnings (see note in `FieldCapabilitiesFilterTests`).  On the other hand verifying that index field data can be loaded is a good test that the field is actually aggregatable?

Reply:
I see, if it isn't too much then I think we should try it. 

It does avoid many deprecation warning when asking field caps for all fields, since it will try to load Index field data for _id field and that triggers deprecation warning (I had to alter a test to take into account these deprecation warnings).